### ISSUE TITLE
Usage with application level singletons?

### ISSUE THREAD

**Issue by Rush** [ngneat/until-destroy#49]
From README:
> Note: The decorator and operator should be used only with component or directive providers. Don't use it with application-level singletons (a.k.a. providedIn: 'root')

Can I ask why? When using live reload in development, `untilDestroyed` is equally useful. I'm using it all the time.

**Comment by arturovt** [comment]
What for? :)

When your tab is reloaded - all memory gets released, there is no need to unsubscribe.

**Comment by NetanelBasal** [comment]
I think he meant hmr.

**Comment by NetanelBasal** [comment]
Note that you're adding extra code (untilDestroyed) to support development mode.

**Comment by Rush** [comment]
Let's say you're working on an app that uses firebase. It's likely you have singleton services that create some subscriptions - and every such subscription holds an active websocket. With HMR the tab does not reload and new instances of the angular app and new instances of services get created. This leads to gradual slowdown and leakage of resources ... I know because I had this problem until I found `untilDestroyed`. It is extra code in non-HMR environment but in my opinion it's totally worth it.

Also, such singleton services made with `untilDestroyed` can easily be converted to component level services.

Additionally singleton services get destroyed in unit tests all the time ...

I think handling proper lifecycle in singleton services is simply a good idea - that's why I got confused by the README. Hopefully the use case is still supported with Ivy.
